### PR TITLE
Add CupertinoPageRoute to Navigation Basics example

### PR DESCRIPTION
**Pull request by miquelbeltran**
_Description of what this PR is changing or adding, and why:_

Following the proposal in https://github.com/flutter/website/issues/2616, this PR adds a section showing to do basic navigation using the Cupertino family widgets.

The example shows how devs can replace the `MaterialPageRoute` with a `CupertinoPageRoute`, as well as the equivalents for `MaterialApp`, `ElevatedButton`, etc. on Cupertino iOS-style widgets.

I have also added a screen recording in gif format to show those users that don't have javascript.

_Issues fixed by this PR (if any):_

- #2616

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/cookbook/navigation/navigation-basics.md [hunk 1]
```diff
@@ -190,8 +190,104 @@ class SecondRoute extends StatelessWidget {
...
+- [`Scaffold`][] replaced by [`CupertinoPageScaffold`][].
+- [`ElevatedButton`][] replaced by [`CupertinoButton`][].
+
+This way, the example follows the current iOS design language.
+
+{{site.alert.secondary}}
+  You don't need to replace all Material widgets with Cupertino 
+  to use [`CupertinoPageRoute`][]
+  since Flutter allows you to mix and match Material and Cupertino widgets
+  depending on your needs
```
Comment (sfshaza2) [inline src/cookbook/navigation/navigation-basics.md:217]:
```suggestion
  depending on your needs.
```

I literally didn't know that!!

**Review by sfshaza2 — COMMENTED** [review]
Hi @miquelbeltran! Is this staged anywhere? I'd like to see how it looks to have both Material and Cupertino on the same page.

**Review by sfshaza2 — APPROVED** [review]
lgtm!

**Comment by miquelbeltran** [comment]
The staging CI job failed, if it fails again I will host it somewhere manually

**Comment by miquelbeltran** [comment]
@sfshaza2 here's a demo: https://mb-flutter-dev-1.web.app/cookbook/navigation/navigation-basics

File: src/cookbook/navigation/navigation-basics.md [hunk 2]
```diff
@@ -190,8 +190,104 @@ class SecondRoute extends StatelessWidget {
...
+In the following example, these widgets have been replaced:
+
+- [`MaterialApp`][] replaced by [`CupertinoApp`].
+- [`Scaffold`][] replaced by [`CupertinoPageScaffold`][].
+- [`ElevatedButton`][] replaced by [`CupertinoButton`][].
+
+This way, the example follows the current iOS design language.
+
+{{site.alert.secondary}}
+  You don't need to replace all Material widgets with Cupertino
```
Comment (parlough) [inline src/cookbook/navigation/navigation-basics.md:214]:
```suggestion
  You don't need to replace all Material widgets with Cupertino versions
```

**Review by parlough — APPROVED** [review]
Cool! I like how you presented this as an alternative after the recipe steps and discussing the differences.

**Review by domesticmouse — APPROVED** [review]

**Comment by atsansone** [comment]
@miquelbeltran : The error is channel limit reached in Firebase. You can merge this if you want.

**Pull request merged**